pyomo/core/tests/unit/test_component_constraint.py

```diff
@@ -381,7 +398,7 @@ def test_equality(self):
         self.assertEqual(c.rhs, 3)
         self.assertEqual(c.equality, True)
 
-        with self.assertRaises(TypeError):
+        with self.assertRaises(ValueError):
             c.rhs = 'a'
```

Comment:
Why is this a ValueError now? I would think that the former TypeError was more appropriate.